### ISSUE TITLE
Control Flow Graph view for other languages

### ISSUE THREAD

**Issue by JohanEngelen** [mattgodbolt/compiler-explorer#574]
I just discovered the control flow graph mode, pretty nice.
How much effort is it to enable that for LDC? Because the backend is LLVM, the asm output is (almost) the same as Clang's.
Cheers!

**Comment by mattgodbolt** [comment]
@cppchedy - any thoughts on this?

**Comment by JohanEngelen** [comment]
https://github.com/mattgodbolt/compiler-explorer/blob/master/lib/cfg.js#L144

Perhaps try this? ;-)
```diff
- if (compiler.includes("clang"))
+ if (compiler.includes("clang") || compiler.includes("ldc"))
```

**Comment by cppchedy** [comment]
didn't have a look at the output but from your quote 

> the asm output is (almost) the same as Clang's.

it should be easy enough, I will investigate more and make changes. 
look at  lib/cfg.js it contains all the predicates in clangX86.

**Comment by JohanEngelen** [comment]
(same for Rust, and other LLVM-backed compilers, I reckon)

**Comment by cppchedy** [comment]
yes I looked at Rust so it's basically the same. you need to provide a predicate for function name. that's the only difference but I will try to work sth for the other compilers soon enough.

**Comment by jaredwy** [comment]
Instead of that, perhaps push it onto the compiler object it self to declare its "type" then we don't have to keep updating that if statement if we want to support a new compiler that supports new features.

**Comment by cppchedy** [comment]
yes! I need to make addition of new compiler/langs easier.

**Comment by JohanEngelen** [comment]
would be cool if you could also show the CFG for LLVM IR.... ;-)

**Comment by cppchedy** [comment]
@JohanEngelen I didn't think about that, and I don't know if it's feasible giving the current setup.

**Comment by JohanEngelen** [comment]
Implemented for LDC in #868
